src/redux/books/booksSlice.js

```diff
@@ -1,40 +1,101 @@
-import { createSlice } from '@reduxjs/toolkit';
+import { createSlice, createAsyncThunk, isRejectedWithValue } from '@reduxjs/toolkit';
+import axios from 'axios';
 
+const appId = 'm5VanncxBlzO2m73NS26';
+const baseUrl = `https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/${appId}/books`;
 const initialState = {
-  bookItems: [
-    {
-      item_id: 'item1',
-      title: 'The Great Gatsby',
-      author: 'John Smith',
-      category: 'Fiction',
-    },
-    {
-      item_id: 'item2',
-      title: 'Anna Karenina',
-      author: 'Leo Tolstoy',
-      category: 'Fiction',
-    },
-    {
-      item_id: 'item3',
-      title: 'The Selfish Gene',
-      author: 'Richard Dawkins',
-      category: 'Nonfiction',
-    },
-  ],
+  bookItems: {},
+  title: '',
+  author: '',
+  error: undefined,
+  isLoading: false,
+  count: 0,
 };
 
+export const fetchBook = createAsyncThunk('books/fetchBook', async () => {
+  try {
+    const response = await axios.get(baseUrl);
+    return response.data;
+  } catch (error) {
+    return isRejectedWithValue(error.response.data);
+  }
+});
+
+export const postBook = createAsyncThunk('books/postBook', async (newBook) => {
+  try {
+    const response = await axios.post(baseUrl, newBook);
+    return response.data;
+  } catch (error) {
+    return isRejectedWithValue(error.response.data);
+  }
+});
+
+export const removeBook = createAsyncThunk('books/removeBook', async (id) => {
+  try {
+    const response = await axios.delete(`${baseUrl}/${id}`);
+    return response.data;
+  } catch (error) {
+    return isRejectedWithValue(error.response.data);
+  }
+});
+
 export const booksSlice = createSlice({
   name: 'books',
   initialState,
   reducers: {
-    addBook: (state, action) => {
-      state.bookItems.push(action.payload);
+    // addBook: (state, action) => {
+    //   state.bookItems.push(action.payload);
+    // },
+    // removeBook: (state, action) => {
+    //   state.bookItems = state.bookItems.filter((book) => book.item_id !== action.payload);
+    // },
```

Comment:
- [ ] Please kindly remove all commented code from your codebase as it pollutes it and also not best practice